src/stream/stream.rs

```diff
@@ -168,3 +231,50 @@ impl<S: futures::Stream> futures::Stream for Take<S> {
         }
     }
 }
+
+#[derive(Debug)]
+pub struct AllFuture<'a, S, F, T>
+where
+    F: FnMut(T) -> bool,
+{
+    stream: &'a mut S,
+    f: F,
+    result: bool,
+    __item: PhantomData<T>,
```

Comment:
I don't know is that a good way to name the `PhantomData` fields, feel free to comment on this!

Reply:
This looks fine!